src/mumble/CoreAudio.h

```diff
@@ -6,50 +6,63 @@
 #ifndef MUMBLE_MUMBLE_COREAUDIO_H_
 #	define MUMBLE_MUMBLE_COREAUDIO_H_
 
+#	include <AudioToolbox/AudioToolbox.h>
+
 #	include "AudioInput.h"
 #	include "AudioOutput.h"
 
-#	include <AudioToolbox/AudioToolbox.h>
-#	include <Carbon/Carbon.h>
-
-#	include "Global.h"
+enum AUElement { OUTPUT = 0, INPUT = 1 };
 
 class CoreAudioSystem : public QObject {
 private:
 	Q_OBJECT
 	Q_DISABLE_COPY(CoreAudioSystem)
 public:
-	static CFStringRef QStringToCFString(const QString &str);
-	static const QHash< QString, QString > getDevices(bool input);
+	static const QHash< QString, QString > getDevices(bool input, bool echo);
 	static const QList< audioDevice > getDeviceChoices(bool input);
 };
 
 class CoreAudioInput : public AudioInput {
 private:
 	Q_OBJECT
 	Q_DISABLE_COPY(CoreAudioInput)
+	// Open HAL AU as input and pass back the output stream description.
+	bool openAUHAL(AudioStreamBasicDescription &streamDescription);
+
+	// Open VoiceProcessingIO AU as input, utilizing macOS's builtin echo cancellation,
+	// and pass back the output stream description.
+	bool openAUVoip(AudioStreamBasicDescription &streamDescription);
+
+	// Initialize input AU with preferred parameters of Mumble
+	bool initializeInputAU(AudioUnit au, AudioStreamBasicDescription &streamDescription, int &actualBufferLength);
```

Comment:
```suggestion
	/// Initialize input AU with preferred parameters of Mumble
	bool initializeInputAU(AudioUnit au, AudioStreamBasicDescription &streamDescription, int &actualBufferLength);
```

Is it intentional that the `AudioUnit` is passed by value instead of reference?

Reply:
Actually, it's a long integer. But I can turn it into a const reference.

Reply:
No need to use a reference for a POD type. The name just suggested an object-like type and therefore I was wondering. But if it is POD then passing by value is fine (and afaik actually preferred)